### PR TITLE
New Cheat Sheet Apache Maven Goodies

### LINKED ISSUE #3818: Suggestion: New Apache Maven Cheat Sheet
Cheat Sheet for maven, JVM Build Tool

------
IA Page:  http://duck.co/ia/view/apache_maven_cheat_sheet 
[Maintainer](http://docs.duckduckhack.com/maintaining/guidelines.html): @Hasil-Sharma

### PR DESCRIPTION
**Pull request by Hasil-Sharma**
## Description of new Instant Answer, or changes

CheatSheet for Apache Maven, Java Build Tool

## Related Issues and Discussions

Fixes #3818

## People to notify

---

Instant Answer Page: https://duck.co/ia/view/apache_maven_cheat_sheet

### CODE REVIEW

**Comment by daxtheduck** [comment]
Metadata placeholder post
            
*If you haven't already, please edit your first comment to include a link to the IA page at the bottom.*
            
---
            
**How to find your IA page**
- New Instant Answer
  -- follow the steps [here](https://duck.co/ia/new_ia) to create your IA page.
              
- Existing Instant Answer
  -- find your IA page from this [list](https://duck.co/ia)

File: share/goodie/cheat_sheets/json/apache-maven.json [hunk 1]
```diff
@@ -0,0 +1,107 @@
...
+        "General": [
+			{
+				"key" : "pom.xml",
+				"val" : "Configuration file which maintains versioning, dependencies and plugins"
+			},
+            {
+                "key": "mvn plugin:target \\[-Doption1 -Doption2\\]",
+                "val": "General command template, to be executed in pom.xml's directory"
+            }
+        ],
```
Comment (amarlearning) [inline share/goodie/cheat_sheets/json/apache-maven.json]: Indentation is not done properly!

Comment (Hasil-Sharma) [inline share/goodie/cheat_sheets/json/apache-maven.json]: Aah ! 😟 my bad ... mixed up indents and spaces ... fixed 😄

**Comment by Hasil-Sharma** [comment]
@amarlearning : Can you please help me in figuring out why this is failing ? I checked with `duckpan test CheatSheets` and it gave no errors.

**Comment by ManrajGrover** [comment]
@Hasil-Sharma Looks like restarting build fixed it.

File: share/goodie/cheat_sheets/json/apache-maven.json (hunk 1 above)
Comment (sahildua2305) [inline share/goodie/cheat_sheets/json/apache-maven.json]: Side note - you can use this [tool](https://duckduckgo.com/?q=json+validate&atb=v44-1__&ia=answer) as well 😉 #ShowOff 💯

Comment (Hasil-Sharma) [inline share/goodie/cheat_sheets/json/apache-maven.json]: Hahaha ... @sahildua2305  self marketing not showoff ... Finally I used your tool to make do 😄

**Comment by Hasil-Sharma** [comment]
Any other change I need to make in this ?

**Comment by amarlearning** [comment]
Rest looks good to me

On 23-Dec-2016 4:11 PM, "Hasil Sharma" <notifications@github.com> wrote:

Any other change I need to make in this ?

—
You are receiving this because you were mentioned.
Reply to this email directly, view it on GitHub
<https://github.com/duckduckgo/zeroclickinfo-goodies/pull/3820#issuecomment-268971324>,
or mute the thread

File: share/goodie/cheat_sheets/json/apache-maven.json (hunk 1 above)
Comment (gautamkrishnar) [inline share/goodie/cheat_sheets/json/apache-maven.json]: Hehehe 😄 ❤️

**Comment by gautamkrishnar** [comment]
@Hasil-Sharma you need not have to open issues before creating PR's. Please stick out to the slack channel if you are not sure whether the IA needs to be developed or not.

**Pull request merged**